src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -254,12 +257,16 @@ export function isNeverType(type: Type): type is NeverType {
     return type.kind === "Intrinsic" && type.name === "never";
 }
 
+export function isEnumMember(type: Type): boolean {
+    return type.kind === "EnumMember";
+}
```

Comment:
We have quite a few functions like this, is it possible that we could make this more strong typed?
Something like
```
export function isKind(type: Type, kind: TypeKind): boolean {
    return type.kind === kind;
}
```
and we define a new type `TypeKind` which collects all the possible values for kind

Reply:
This method could return `type is EnumMember`

Reply:
Good suggestion. But now it is not enough to check the kind to identify if it is a special type. And now `isKind` cannot replace is***. And the usage of `isKind` is not high. We will plan to implement this in future if needed.